Log.c: In function 'LogFileAppendTempLog':
Log.c:452:31: error: 'LOCK_EX' undeclared (first use in this function)
  452 |         STREAMLock(LogFile->S,LOCK_EX);
      |                               ^~~~~~~
Log.c:452:31: note: each undeclared identifier is reported only once for each function it appears in
Log.c:460:31: error: 'LOCK_UN' undeclared (first use in this function)
  460 |         STREAMLock(LogFile->S,LOCK_UN);
      |                               ^~~~~~~
...
In file included from Process.c:29:
.../include/sys/prctl.h:88:8: error: redefinition of 'struct prctl_mm_map'
   88 | struct prctl_mm_map {
      |        ^~~~~~~~~~~~
In file included from Process.c:27:
.../include/linux/prctl.h:133:8: note: originally defined here
  133 | struct prctl_mm_map {
      |        ^~~~~~~~~~~~

---

--- a/libUseful-bundled/Log.c
+++ b/libUseful-bundled/Log.c
@@ -2,6 +2,9 @@
 #include "Time.h"
 #include "FileSystem.h"
 #include <syslog.h>
+#ifndef __GLIBC__
+#include <sys/file.h> /* musl LOCK_** */
+#endif
 
 
 //When logs are used by 'child processes', it's only the 'parent' process that should be
--- a/libUseful-bundled/OAuth.c
+++ b/libUseful-bundled/OAuth.c
@@ -7,6 +7,9 @@
 #include "Users.h"
 #include "Entropy.h"
 #include "URL.h"
+#ifndef __GLIBC__
+#include <sys/file.h> /* musl LOCK_** */
+#endif
 
 static ListNode *OAuthTypes=NULL;
 static ListNode *OAuthKeyChain=NULL;
--- a/libUseful-bundled/Process.c
+++ b/libUseful-bundled/Process.c
@@ -23,7 +23,9 @@
 #include <sys/file.h>
 
 #ifdef HAVE_PRCTL
-#include <linux/prctl.h>  /* Definition of PR_* constants */
+# ifdef __GLIBC__
+#  include <linux/prctl.h>  /* Definition of PR_* constants */
+# endif
 #include <sys/prctl.h>
 #endif
 
--- a/upload.c
+++ b/upload.c
@@ -1,6 +1,9 @@
 #include "upload.h"
 #include "server.h"
 #include "Events.h"
+#ifndef __GLIBC__
+#include <sys/file.h> /* musl LOCK_** */
+#endif
 
 #define UPLOAD_FAILED -1
 #define UPLOAD_DONE 1
